product/dialogs/miq_dialogs/miq_provision_google_dialogs_template.yaml

```diff
@@ -205,14 +205,18 @@
           :notes_display: :show
         :vm_name:
           :description: Instance Name
-          :required_method: :validate_vm_name
+          :required_method:
+            - :validate_vm_name
+            - :validate_regex
+          :required_regex: !ruby/regexp /\A(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)\Z/
+          :required_regex_fail_details: The name must be composed only of lower-case English alphabet characters (a-z), numbers (0-9), and dashes (-). The first character must be a lower-case English alphabet character, and the last character must not be a dash.
```

Comment:
@selmanj @agrare The only thing to note here is that users could specific an enumeration pattern (`$n{#}`) as part of the VM name and this would prevent that.  I do not believe it is a common use-case and users could update the dialog if it is required to support it.

Here's an example where automate applies it based on provisioning more than one VM: [db/fixtures/ae_datastore/ManageIQ/Cloud/VM/Provisioning/Naming.class/__methods__/vmname.rb#L53](https://github.com/ManageIQ/manageiq/blob/355f903573aa16214c5a1b938f564df03d6a8228/db/fixtures/ae_datastore/ManageIQ/Cloud/VM/Provisioning/Naming.class/__methods__/vmname.rb#L53)